### PR TITLE
Fix leaving swarm if node is in error state

### LINKED ISSUE #28490: Joining swarm a second time as a worker after promotion made the node a manager
I tried to add a second manager using a dind setup. I added the new node as a worker and then promoted it. I forgot to set the listen address to something that would not conflict. The manager did not start successfully, and I tried to remove the node from the swarm and try again. When I rejoined the cluster, the node was a manager even though I used the worker join token. It seems to have retained the same certificate as before.

```
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker -H unix:///var/run/docker2.sock swarm join --token <worker token> 172.17.0.3:2377
This node joined a swarm as a worker.
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker node ls
ID                           HOSTNAME      STATUS  AVAILABILITY  MANAGER STATUS
6y2eiflcajy8ncnjfoi0joew4    56cdb93b9fa5  Ready   Active
dzd9a6s7dsggoe5nih23bcdtg *  56cdb93b9fa5  Ready   Active        Leader
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker node promote 6y2eiflcajy8ncnjfoi0joew4
Node 6y2eiflcajy8ncnjfoi0joew4 promoted to a manager in the swarm.
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker -H unix:///var/run/docker2.sock swarm leave
Error response from daemon: listen tcp 0.0.0.0:2377: bind: address already in use
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker -H unix:///var/run/docker2.sock swarm leave --force
Error response from daemon: This node is not part of a swarm
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker node ls
ID                           HOSTNAME      STATUS  AVAILABILITY  MANAGER STATUS
6y2eiflcajy8ncnjfoi0joew4    56cdb93b9fa5  Down    Active
dzd9a6s7dsggoe5nih23bcdtg *  56cdb93b9fa5  Ready   Active        Leader
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker -H unix:///var/run/docker2.sock swarm join --token <worker token>  --listen-addr 172.17.0.3:2378   172.17.0.3:2377
This node joined a swarm as a manager.
root@56cdb93b9fa5:/go/src/github.com/docker/docker# docker node ls
ID                           HOSTNAME      STATUS  AVAILABILITY  MANAGER STATUS
6y2eiflcajy8ncnjfoi0joew4    56cdb93b9fa5  Ready   Active        Reachable
dzd9a6s7dsggoe5nih23bcdtg *  56cdb93b9fa5  Ready   Active        Leader
```

```
# docker version
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 21:39:14 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 21:39:14 2016
 OS/Arch:      linux/amd64
```

cc @tonistiigi @LK4D4 @diogomonica

### PR DESCRIPTION
**Pull request by tonistiigi**
fixes #28490

Manager is restored in that issue because there never was a successful `swarm leave` command and state was not removed. This fixes `swarm leave` when the node in between restarts.

`Error response from daemon: listen tcp 0.0.0.0:2377: bind: address already in use` for the first `swarm leave` in #28490 comes from https://github.com/docker/docker/blob/1.12.x/vendor/src/github.com/docker/swarmkit/agent/node.go#L330 . Currently vendored version of swarmkit doesn't return node error on stop.

@aaronlehmann 

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

**Comment by vieux** [comment]
@tonistiigi for 1.13 ?

File: daemon/cluster/cluster.go [hunk 1]
```diff
@@ -626,10 +626,6 @@ func (c *Cluster) stopNode() error {
 		return nil
 	}
 	c.stop = true
-	if c.cancelDelay != nil {
-		c.cancelDelay()
-		c.cancelDelay = nil
-	}
```
Comment (aaronlehmann) [inline daemon/cluster/cluster.go]: Trying to understand moving this to `Cleanup`. Doesn't the call to `stopNode` from `Init` still need it?

**Comment by vieux** [comment]
ping @tonistiigi

**Comment by tonistiigi** [comment]
will make a different(and much bigger) refactor for this.

**Pull request closed without merging**